packages/@react-stately/selection/src/useMultipleSelectionState.ts

```diff
@@ -59,15 +59,22 @@ export function useMultipleSelectionState(props: MultipleSelection): MultipleSel
     get childFocusStrategy() {
       return childFocusStrategyRef.current;
     },
-    setFocusedKey(k, childFocusStrategy = 'first') {
+    setFocusedKey(k, childFocusStrategy = 'first', callbackSet) {
       focusedKeyRef.current = k;
       childFocusStrategyRef.current = childFocusStrategy;
-      setFocusedKey(k);
+
+      if (callbackSet) {
+        for (let handler of callbackSet) {
+          handler(k);
+        }
+      }
     },
     selectedKeys,
     setSelectedKeys,
     disabledKeys: disabledKeysProp
-  };
+  }), [selectionMode, disallowEmptySelection, disabledKeysProp, selectedKeys]);
```

Comment:
Omitted `setSelectedKeys` because it was causing ComboBox to get a new SelectionManager every render which broke aria-activedescendant updates in some cases. Not entirely sure why the state here wasn't stable when I included `setSelectedKeys` in the dep array...